storage/Filesystems/BlkFilesystemImpl.cc

```diff
@@ -232,28 +232,31 @@ namespace storage
 
 	ResizeInfo resize_info(true, min_size(), max_size());
 
+	unsigned long long blk_device_size = get_blk_device()->get_size();
+
 	// Even if shrink or grow is not supported it is still possible to
 	// shrink or grow to the original size.
 
 	if (!supports_shrink())
 	{
-	    unsigned long long original_size = get_blk_device()->get_size();
-
-	    resize_info.min_size = original_size;
+	    resize_info.min_size = blk_device_size;
 	}
 	else
 	{
 	    // Assume that the min-size, e.g. for superblocks, metadata and
 	    // journal, is needed additional to the used-size.
 
 	    resize_info.min_size += used_size_on_disk();
+
+	    // But the min-size must never be smaller than the blk device
+	    // size.
+
+	    resize_info.min_size = min(resize_info.min_size, blk_device_size);
```

Comment:
According to the comment, should it be `max` here?

Reply:
The comment is wrong. I will fix that.